src/main/java/org/folio/service/budget/BudgetService.java

```diff
@@ -237,7 +175,91 @@ public Future<Void> closeBudgets(LedgerFiscalYearRollover rollover, DBConn conn)
         getFullTableName(conn.getTenantId(), BUDGET_TABLE), getFullTableName(conn.getTenantId(), FUND_TABLE),
         rollover.getLedgerId(), rollover.getFromFiscalYearId());
 
-    return budgetDAO.updateBatchBudgets(sql, conn)
+    return budgetDAO.updateBatchBudgetsBySql(sql, conn)
+      .mapEmpty();
+  }
+
+  public Future<List<Budget>> getBudgetsByFiscalYearIdsAndFundIdsForUpdate(
+      Map<String, Set<String>> fiscalYearIdToFundIds, DBConn conn) {
+    StringBuilder sb = new StringBuilder();
+    sb.append("SELECT jsonb FROM ")
+      .append(getFullTableName(conn.getTenantId(), BUDGET_TABLE))
+      .append(" WHERE ");
+    Iterator<String> fiscalYearIterator = fiscalYearIdToFundIds.keySet().iterator();
+    while (fiscalYearIterator.hasNext()) {
+      String fiscalYearId = fiscalYearIterator.next();
+      Set<String> fundIds = fiscalYearIdToFundIds.get(fiscalYearId);
+      sb.append("(jsonb->>'fiscalYearId' = '")
```

Comment:
The budget table has an foreign key for fiscalYearId and RMB automatically creates indexes for foreign keys.
If possible we need to use them, the same also for fundId

Reply:
Good idea.